mrview & shview version mismatch on dev

On the current `dev` branch, I did a full delete of my `cmake` build directory, a `ccache -C`, and re-built in a new directory. Commands `mrview` and `shview` nevertheless fail to execute due to mismatch between `git` versions reported for the executable vs. the shared library.

```
robertes@9520l-003953-l:~/src/mrtrix3$ ldd ./build_cmake/bin/mrview
	libmrtrix-gui.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/src/libmrtrix-gui.so (0x00007fe8e9400000)
	libmrtrix-core.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/core/libmrtrix-core.so (0x00007fe8e9000000)
	libmrtrix-headless.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/src/libmrtrix-headless.so (0x00007fe8e7600000)
	...
robertes@9520l-003953-l:~/src/mrtrix3$ ./build_cmake/bin/mrview
mrview: [ERROR] executable was compiled for a different version of the MRtrix3 library!
mrview: [ERROR]   mrview version: 3.0.4-708-g895628e0
mrview: [ERROR]   library version: 3.0.4-623-g4f3836b9-dirty
mrview: [ERROR] Running ./build again may correct error
```

In addition to resolving whatever is causing the conflict, the error message also needs to be updated to reflect the `cmake` transition.

I've wiped my `PATH` of all traces of MRtrix3, ensured `LD_LIBRARY_PATH` is empty, wiped compiler cache, deleted build directory, configured and built:

```
robertes@9520l-003953-l:~/src/mrtrix3$ ./build_cmake/bin/mrinfo -version
== mrinfo 3.0.4-623-g4f3836b9-dirty ==
...
robertes@9520l-003953-l:~/src/mrtrix3$ ldd ./build_cmake/bin/mrinfo
	libmrtrix-headless.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/src/libmrtrix-headless.so (0x00007fc9fd800000)
	libmrtrix-core.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/core/libmrtrix-core.so (0x00007fc9fd400000)
...
robertes@9520l-003953-l:~/src/mrtrix3$ ./build_cmake/bin/shview -version
shview: [ERROR] executable was compiled for a different version of the MRtrix3 library!
shview: [ERROR]   shview version: 3.0.4-711-g81d8576d-dirty
shview: [ERROR]   library version: 3.0.4-623-g4f3836b9-dirty
shview: [ERROR] Running ./build again may correct error
robertes@9520l-003953-l:~/src/mrtrix3$ ldd ./build_cmake/bin/shview
	libmrtrix-gui.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/src/libmrtrix-gui.so (0x00007ff5bc600000)
	libmrtrix-core.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/core/libmrtrix-core.so (0x00007ff5bc200000)
	libmrtrix-headless.so => /home/unimelb.edu.au/robertes/src/mrtrix3/build_cmake/src/libmrtrix-headless.so (0x00007ff5ba800000)
...
robertes@9520l-003953-l:~/src/mrtrix3$ git describe --abbrev=8 --dirty --always
3.0.4-711-g81d8576d-dirty
```

So it seems to me that *only* the GUI executables are capturing the correct version identifier; the headless binaries and the shared libraries are all capturing some *incorrect* version, but because it's the *same* incorrect version, execution there is successful.

--- a/core/app.cpp
+++ b/core/app.cpp
@@ -1101,7 +1101,9 @@ void init(int cmdline_argc, const char *const *cmdline_argv) {
     Exception E("executable was compiled for a different version of the MRtrix3 library!");
     E.push_back(std::string("  ") + NAME + " version: " + executable_uses_mrtrix_version);
     E.push_back(std::string("  library version: ") + mrtrix_version);
-    E.push_back("Running ./build again may correct error");
+    E.push_back("You may need to erase files left over from prior MRtrix3 versions;");
+    E.push_back("eg. core/version.cpp; src/exec_version.cpp");
+    E.push_back(", and re-configure cmake");
     throw E;
   }
 

